### PR TITLE
Extract and store icons for FxOS Add-ons, and expose them in the API (bug 1217441)

### PR DESCRIPTION
**Pull request by diox**
https://bugzilla.mozilla.org/show_bug.cgi?id=1217441

### CODE REVIEW

File: mkt/webapps/tests/test_crons.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def test_by_webapp(self, sign_mock):
         v1 = self.app.current_version
         call_command('sign_apps', webapps=str(v1.pk))
         file1 = v1.all_files[0]
-        assert sign_mock.called_with(((file1.file_path,
+        sign_mock.assert_called_with(((file1.file_path,
```
Comment (diox) [inline mkt/webapps/tests/test_crons.py]: drive by fix, unrelated.

File: mkt/extensions/migrations/0017_author_field_length.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (diox) [inline mkt/extensions/migrations/0017_author_field_length.py]: missing migration.

File: mkt/extensions/migrations/0018_change_default_ordering.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
```
Comment (diox) [inline mkt/extensions/migrations/0018_change_default_ordering.py]: missing migration.

File: mkt/extensions/tests/test_tasks.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        self.version = self.extension.versions.create(manifest={
+            'icons': {
+                '128': '/path/to/icon.png'
+            }
+        }, version='0.1')
+
+    @mock.patch('mkt.extensions.tasks.save_icon')
+    @mock.patch('mkt.extensions.tasks.ZipFile')
+    @mock.patch('mkt.extensions.tasks.private_storage')
+    def _test(self, private_storage_mock, ZipFile_mock, save_icon_mock,
```
Comment (washort) [inline mkt/extensions/tests/test_tasks.py]: these kind of mock-heavy internal-behaviour tests make me a little nervous; they definitely get the job done but they're the most exasperating thing when it's time to refactor/change backends. I don't have a suggestion for what specifically to do instead here though.

Comment (diox) [inline mkt/extensions/tests/test_tasks.py]: I agree - I spent a lot of time running around trying to find a better way of doing this, and this is the best I could come up with. I tried to keep the calls and therefore the mocks simple and heavily document + test the various cases to balance the awkwardness of the mocks.

**Comment by washort** [comment]
r+

**Pull request merged**